neuraxle/metaopt/random.py

```diff
@@ -581,17 +582,53 @@ def __init__(
             wrapped=None,
             n_iter: int = 10,
             higher_score_is_better: bool = True,
-            validation_technique: BaseCrossValidationWrapper = KFoldCrossValidationWrapper(),
+            validation_technique: BaseValidation = KFoldCrossValidationWrapper(),
             refit=True,
     ):
         if wrapped is not None:
             MetaStepMixin.__init__(self, wrapped)
         BaseStep.__init__(self)
         self.n_iter = n_iter
         self.higher_score_is_better = higher_score_is_better
-        self.validation_technique: BaseCrossValidationWrapper = validation_technique
+        self.validation_technique: BaseValidation = validation_technique
         self.refit = refit
 
+    def _fit_transform_data_container(self, data_container, context):
+        fitted_self = self._fit_data_container(data_container, context)
+        best_model_predictions_data_container = self._transform_data_container(data_container, context)
+        return fitted_self, best_model_predictions_data_container
+
+    def _fit_data_container(self, data_container, context):
+        started = False
+        best_hyperparams = None
+
+        for _ in range(self.n_iter):
+
+            step = copy.copy(self.wrapped)
+
+            new_hyperparams = step.get_hyperparams_space().rvs()
+            step.set_hyperparams(new_hyperparams)
+
+            step: BaseValidation = copy.copy(self.validation_technique).set_step(step)
+
+            step = step.handle_fit(data_container, context)
+            score = step.scores_mean
+
+            if not started or self.higher_score_is_better == (score > self.score):
+                started = True
+                self.score = score
+                self.best_validation_wrapper_of_model = copy.copy(step)
+                print('score: {}'.format(score))
+                best_hyperparams = new_hyperparams
+                print('best_hyperparams: \n{}\n'.format(best_hyperparams))
+
+        self.best_validation_wrapper_of_model.wrapped.set_hyperparams(best_hyperparams)
+
+        if self.refit:
+            self.best_model = self.best_validation_wrapper_of_model.wrapped.handle_fit(data_container, context)
```

Comment:
I think here we should take a new copy of the model and assign it the best found parameters. And then completely retrain it. It seems here like there is a mistake like if we continue fitting the found model. Continuing the fit wouldn't be the good thing to do.